### ISSUE TITLE
Add a conversion to/from BigDecimal and String

### ISSUE THREAD

**Issue by DefiantCatgirl** [MorphiaOrg/morphia#1524]
In our project due to using Spring's MongoRepository for years (and gradually updating from early MongoDB versions) `BigDecimal` fields in entities were and still are being saved as strings, and in Morphia 1.* the strings were correctly converted back to `BigDecimal` when reading entities from the database. Is it possible when using 2.0 to add some kind of converter (codec?) that would do the same if it encounters a string in the database but a `BigDecimal` as the type of the corresponding field in the class? Ideally just to Morphia, without adding it to the underlying MongoClient as it may be also used elsewhere.

Currently 2.0 is throwing `java.lang.IllegalArgumentException: Can not set java.math.BigDecimal field com.foo.Bar.value to java.lang.String` in this case.

(collections read & saved only by Morphia _(1.*, haven't tried with 2.0 yet)_ are using NumberDecimal as would be expected, but a migration of all of the legacy data is beyond troublesome at the moment, especially considering MongoRepository still has to be used here and there for legacy reasons...)

    Server Version: 3.6
    Driver Version: 4.0.5
    Morphia Version: 2.0.2

**Comment by evanchooly** [comment]
Yes, that's entirely possible.  I consider it a bug that 2.0 broke that.  Ultimately what changed was how those conversions were defined and in that process that one got dropped.  I'll work on a bug fix (probably for 2.1 since it's so close unless you *really* want/need a 2.0.2)) but here's how they work in 2.0.  (Note:  this API is marked as `internal` so the details may change over time.)

So here's how Morphia defines a conversion:

```java
dev.morphia.mapping.codec.Conversions#register(java.lang.Class<S>, java.lang.Class<T>, java.util.function.Function<S,T>)
```

It basically takes source and target types and a lambda to do that conversion.  You can see several examples in the `Conversions` class.  I'll add one for BigDecimal so that it comes stock.  You can try adding one yourself in the meantime so you don't have to wait.  But I wouldn't rely on it for too long once it lands in Morphia.  As I mentioned that API is `internal` and though I don't have any plans, immediate or not`, to change that API, something might come up in the future requiring a tweak that would break your code.

I hope this help.

**Comment by evanchooly** [comment]
`BigDecimal` fields will still *save* as `Decimal128` per the driver's codec, but any existing string values will be converted to `BigDecimal` on loads.

If you need them to save as Strings, you'll need a custom codec (you can borrow the driver's codec for inspiration) but this should help with your loading issues.
